File: Luna.Models.Tasks.Blank/Card/CardBlank.cs
namespace Luna.Models.Tasks.Blank.Card;

public class CardBlank
{
	public String Header { get; set; } = null!;

	public String? Content { get; set; }

	public String? Description { get; set; }

	public Guid CardTypeId { get; set; }

	public Guid StatusId { get; set; }

	public IEnumerable<Guid> TagIds { get; set; } = new List<Guid>();

	public IEnumerable<Guid> UserIds { get; set; } = new List<Guid>();

	public Guid PageId { get; set; }

	public DateTime? Deadline { get; set; }

	public Guid? PreviousCardId { get; set; }
}

File: luna/src/models/card/blank/cardBlank.ts
export default class CardBlank {
    constructor(
        header: string,
        content: string | null,
        description: string | null,
        cardTypeId: string,
        pageId: string | null,
        deadline: string | null,
        previousCardId: string | null
    ) {
        this.header = header;
        this.content = content;
        this.description = description;
        this.cardTypeId = cardTypeId;
        this.pageId = pageId;
        this.deadline = deadline;
        this.previousCardId = previousCardId;
    }

    public header: string;
    public content: string | null;
    public description: string | null;
    public cardTypeId: string;
    public pageId: string | null; // only for create
    public deadline: string| null;
    public previousCardId: string | null;
}